lua/apisix/http/router/radixtree_host_uri.lua

```diff
@@ -0,0 +1,176 @@
+-- Copyright (C) bobo pan
+
+local require = require
+local router = require("resty.radixtree")
+local core = require("apisix.core")
+local plugin = require("apisix.plugin")
+local ipairs = ipairs
+local type = type
+local str_reverse = string.reverse
+local error = error
+local user_routes
+local cached_version
+local only_uri_routes = {}
+local only_uri_router
+local host_uri_routes = {}
+local host_uri_router
+
+
+local _M = {version = 0.1}
+
+
+local function create_only_uri_router()
+    local routes = plugin.api_routes()
+    core.log.error("routes", core.json.encode(routes))
+    for _, route in ipairs(routes) do
+        if type(route) == "table" then
+            core.table.insert(only_uri_routes, {
+                paths = route.uris or route.uri,
+                handler = route.handler,
+                method = route.methods,
+            })
+        end
+    end
+    only_uri_router = router.new(only_uri_routes)
+    return true
+end
+
+
+local function add_host_uri_routes(paths, hosts, route)
+    core.log.info("route rule: ", paths)
+    core.table.insert(host_uri_routes, {
+        paths = paths,
+        methods = route.value.methods,
+        hosts = hosts,
+        remote_addrs = route.value.remote_addrs or route.value.remote_addr,
+        vars = route.value.vars,
+        handler = function (api_ctx)
+            api_ctx.matched_params = nil
+            api_ctx.matched_route = route
+        end
+    })
+end
+
+
+local function push_radixtree_host_router(route)
+    if type(route) ~= "table" then
+        return
+    end
+
+    local host = route.value.host
+    local hosts = route.value.hosts
+    local uri = route.value.uris or route.value.uri
+    if (not host or type(host) ~= "string") and (not hosts or type(hosts) ~= "table") then
+        core.table.insert(only_uri_routes, {
+            paths = route.value.uris or route.value.uri,
+            method = route.value.methods,
+            hosts = hosts or host,
+            remote_addrs = route.value.remote_addrs or route.value.remote_addr,
+            vars = route.value.vars,
+            handler = function (api_ctx)
+                api_ctx.matched_params = nil
+                api_ctx.matched_route = route
+            end
+        })
+        return
+    end
+    --more hosts
+    if hosts and type(hosts) == "table" then
+        for _, host_path in ipairs(hosts) do
+            local paths = "/" .. str_reverse(host_path)
```

Comment:
need to check the `host_path` first, it maybe a wildcard host.

Reply:
and if we do not support wildcard host name now, we do not need to call `str_reverse`.